src/components/utils/src/threads/thread_qt.cc

```diff
@@ -61,6 +61,9 @@ void* Thread::threadFunc(void* arg) {
   threads::Thread* thread = static_cast<Thread*>(arg);
   DCHECK(thread);
 
+  // Sets thread id in order to be able to check that id on thread joining
+  thread->set_thread_handle(QThread::currentThread);
```

Comment:
@dev-gh you have to protect `handle_` variable, so move it down after lock Acquire